src/stretchy.js

```diff
@@ -0,0 +1,179 @@
+    overbracket : [0.3, 0, "overbracket"],
+    underbracket : [0.3, 0, "underbracket"],
+    overleftrightarrow : [0.334, 0, "leftrightarrow"],
+    underleftrightarrow : [0.334, 0, "leftrightarrow"],
+    xleftrightarrow : [0.261, 0.261, "xleftrightarrow"],
+    Overrightarrow : [0.56, 0, "doublerightarrow"],
+    xLeftarrow : [0.28, 0.28, "doubleleftarrow"],
+    xRightarrow : [0.28, 0.28, "doublerightarrow"],
+    xLeftrightarrow : [0.28, 0.28, "doubleleftrightarrow"],
+    overleftharpoon : [0.334, 0, "leftharpoon"],
+    overrightharpoon : [0.334, 0, "rightharpoon"],
+    xleftharpoonup : [0.261, 0.261, "xleftharpoon"],
+    xrightharpoonup : [0.261, 0.261, "xrightharpoon"],
+    xhookleftarrow : [0.261, 0.261, "hookleftarrow"],
+    xhookrightarrow : [0.261, 0.261, "hookrightarrow"],
+    overlinesegment : [0.334, 0, "linesegment"],
+    underlinesegment : [0.334, 0, "linesegment"],
+    xmapsto : [0.261, 0.261, "mapsto"],
+    xrightharpoondown : [0.261, 0.261, "xrightharpoondown"],
+    xleftharpoondown : [0.261, 0.261, "xleftharpoondown"],
+    xrightleftharpoons : [0.358, 0.358, "rightleftharpoons"],
+    xleftrightharpoons : [0.358, 0.358, "leftrightharpoons"],
+    overgroup : [0.262, 0, "overgroup"],
+    undergroup : [0.262, 0, "undergroup"],
+    xtwoheadleftarrow : [0.167, 0.167, "twoheadleftarrow"],
+    xtwoheadrightarrow : [0.167, 0.167, "twoheadrightarrow"],
+    xLongequal : [0.167, 0.167, "longequal"],
+    xtofrom : [0.264, 0.264, "tofrom"],
+};
+
+const svgSpan = function(group, options) {
+    // Create a span with class(es) that refer to the background-image
+    // and/or the mask-image.
+    const label = group.value.label.substr(1);
+    let height = 0;
+    let depth = 0;
+    const classArray = ["stretchy"];
+    let fileName = "";
+
+    if (utils.contains(["widehat", "widetilde", "undertilde"], label)) {
+        // There are four SVG images available for each function.
+        // Choose a taller image when there are more characters.
+        const numChars = group.value.value.length;
+        if (numChars > 5) {
+            height = 0.34;
+            fileName = (label === "widehat" ? "widehat" : "tilde") + "4";
+        } else {
+            const imgIndex = [1, 1, 2, 2, 3, 3][numChars];
+            if (label === "widehat") {
+                height = [0, 0.16, 0.23, 0.23, 0.28, 0.28][numChars];
+                fileName = "widehat" + imgIndex;
+            } else {
+                height = [0, 0.15, 0.195, 0.195, 0.26, 0.26][numChars];
+                fileName = "tilde" + imgIndex;
+            }
+        }
+    } else {
+        const imgData = katexImagesData[label];
+        height = imgData[0];
+        depth = imgData[1];
+        fileName = imgData[2];
+        if (label.substr(0, 1) === "x") {
+            classArray.push("x-arrow");     // Lengthen the arrow via padding.
+        }
+    }
+
+    let node;
+    if (utils.contains(["overbracket", "underbracket"], label)) {
```

Comment:
The preceding `if/else-if` block will select a class and set a default height. So it is necessary. 

The code within the `if (utils.contains(["overbracket", "underbracket"], label)) {` block is there for cases where the author has used optional arguments to set custom height and line weight.